### ISSUE TITLE
Stacked mobs become invunerable when stuck in death animation [WildStacker]

### ISSUE THREAD

**Issue by dro1d-89o** [GC-spigot/AdvancedEnchantments#2558]
### Describe the bug

When using a custom enchanted sword to kill stacks of mobs the killed mob of a stack occasionally gets stuck in the death animation (about halfway of falling over) and becomes invulnerable. 

Below is an image of it happening, as well as the sword i am using. The mob stays like that and is invulnerable, if the player rejoins it goes back to a standing animation but is still invulnerable.

[Link to WildStacker issue](https://github.com/BG-Software-LLC/WildStacker/issues/405)

### How to reproduce

Toss a bunch of stacked entities into a 1 block space and start killing them, should happen soon enough
(seems to happen more often when multiple stacks of stacked entities exist)

### Screenshots / Videos

![image](https://user-images.githubusercontent.com/62203146/147303175-a328b930-dfdf-4d80-85a7-9eb34add297b.png)
![image](https://user-images.githubusercontent.com/62203146/147342763-50fdd802-a7d9-4f82-98fd-f4f2d078122c.png)

### "/ae plinfo" link

https://paste.md-5.net/uyibuzorus

**Comment by OmerBenGera** [comment]
My guess is that AE's enchantments deal more damage than usual, causing to a weird conflict.
Maybe GC can tell me how damage is dealt to entities so I can try and patch it?
~ It's probably a fault of WS and not AE ~

**Comment by GC-spigot** [comment]
> My guess is that AE's enchantments deal more damage than usual, causing to a weird conflict. Maybe GC can tell me how damage is dealt to entities so I can try and patch it? ~ It's probably a fault of WS and not AE ~

`LivingEntity#damage(damageAmount, player)`

**Comment by OmerBenGera** [comment]
> > My guess is that AE's enchantments deal more damage than usual, causing to a weird conflict. Maybe GC can tell me how damage is dealt to entities so I can try and patch it? ~ It's probably a fault of WS and not AE ~
> 
> `LivingEntity#damage(damageAmount, player)` that's about it, as soon as enchant is activated, that is how we deal damage to mobs.

And you deal this damage inside EntityDamageEvent, right?

**Comment by GC-spigot** [comment]
> > > My guess is that AE's enchantments deal more damage than usual, causing to a weird conflict. Maybe GC can tell me how damage is dealt to entities so I can try and patch it? ~ It's probably a fault of WS and not AE ~
> > 
> > 
> > `LivingEntity#damage(damageAmount, player)` that's about it, as soon as enchant is activated, that is how we deal damage to mobs.
> 
> And you deal this damage inside EntityDamageEvent, right?

Yeah, EntityDamageEvent -> looks through enchants -> activates enchant -> runs damage effect. Month ago it was in a runnable, could be causing this issue, but has recently changed to activate inside the event. OP should try updating @dro1d-89o try updating please then let us know.

**Comment by OmerBenGera** [comment]
> > > > My guess is that AE's enchantments deal more damage than usual, causing to a weird conflict. Maybe GC can tell me how damage is dealt to entities so I can try and patch it? ~ It's probably a fault of WS and not AE ~
> > > 
> > > 
> > > `LivingEntity#damage(damageAmount, player)` that's about it, as soon as enchant is activated, that is how we deal damage to mobs.
> > 
> > 
> > And you deal this damage inside EntityDamageEvent, right?
> 
> Yeah, EntityDamageEvent -> looks through enchants -> activates enchant -> runs damage effect. Month ago it was in a runnable, could be causing this issue, but has recently changed to activate inside the event. OP should try updating @dro1d-89o try updating please then let us know.

Try making it as a runnable, it might fix it.
If it does, I will look into what can be done regarding this.
